### DISCUSSION TITLE
Issues with CPDS maybe? Bug? Not sure.

### DISCUSSION THREAD

**Discussion by ILikeToasters** [lllyasviel/Fooocus#3334] [Q&A] [answered]
I am still trying to narrow down the issue I am having but I did make a video showing what I am running into since It is hard to describe. Not sure it is specific to the update but have not had issues until today and have been doing these images recently a lot without issue. Hopefully I am not missing something obvious but would not be the first time.
Video: https://youtu.be/Y3TeNWx016I
These are the example images I was using for CPDS. 
![Untitled (32)](https://github.com/user-attachments/assets/7be3bbea-6d37-420a-80fe-95bbb5c5d87f)
![Untitled (37)](https://github.com/user-attachments/assets/40c19855-1396-4114-a5b9-96ddf1deddf2)
![Untitled](https://github.com/user-attachments/assets/bdd4936c-a0b0-4038-80fe-8762602a99cd)

**Comment by ILikeToasters** [comment]
Further testing if I convert the slanted blue text to a jpg and use it there is no issue.

**Comment by mashb1t** [comment]
Steange, I've only updated the docs today, the last bugfix (metadata reading from jpeg/jpg/webp exif) was 4 days ago. See https://github.com/lllyasviel/Fooocus/commits/main/.

**Reply by ILikeToasters** [reply]
Yeah I can't be 100% sure when it may have started happening since it only happens with certain images. But twice today I have had it not see an image with CPDS and the other time was when inpainting. I did have someone else test and they ran into a similar issue.

**Reply by mashb1t** [reply]
Please create a new issue if you can narrow it down and can reproduce it.

**Reply by ILikeToasters** [reply]
Yeah I am trying to but it is not being consistent. Not sure if you got a chance to check the video since you might see something obvious. It does spit out some errors I found as shown in the video when using the debug preprocessors.

**Reply by mashb1t** [reply]
Thanks, will try to replicate it tomorrow (currently 23:19 CEST)

**Reply by ILikeToasters** [reply]
No problem. I will continue to try and narrow things down but will say if you test with the above images you should get the same results since I just downloaded them to verify they did not get changed when uploading and started working. It is really odd.

**Reply by mashb1t** [reply, marked as answer]
I've been able to debug the attached images. Depending on your target aspect ratio and resolution Fooocus will crop the mask and CPDS then sometimes isn't able to correctly recognize the letters.
See https://github.com/lllyasviel/Fooocus/blob/774c14706bd780a61b2d4d968be5bdb6afe35679/modules/async_worker.py#L410-L413

**Reply by ILikeToasters** [reply]
But is that even the case when the images are the same size as the aspect ratio being generated? Mostly just curious since I can just convert to jpg when I run into that but just odd that I never have before and that just converting to jpg solves it.

**Reply by mashb1t** [reply]
it's so strange, for some resolutions and image file formats it works, for some it doesn't (reproducible)

**Reply by ILikeToasters** [reply]
I do feel pretty confident it is a new issue since I have not run into it before and I do a lot of these images and now I have run into it several times.
